src/utils/ObjectUtils.ts

```diff
@@ -75,3 +79,42 @@ export function groupObjectCountsByProp(objects: any[], getProp: (obj: any) => s
 export function filterFalsyKeys(obj: Record<string | number, unknown>): Record<string | number, unknown> {
   return Object.fromEntries(Object.entries(obj).filter(([, v]) => v));
 }
+
+/**
+ * Deletes keys from an object and returns new copy of object without ignored keys
+ * @param ignoredKeys
+ * @param obj
+ * @returns Objects with ignored keys removed
+ */
+function deleteIgnoredKeys(ignoredKeys: string[], obj: Record<string, unknown>) {
+  if (!isDefined(obj)) {
+    return;
+  }
+  const newObj = { ...obj };
+  for (const key of ignoredKeys) {
+    delete newObj[key];
+  }
+  return newObj;
+}
+
+export function compareResultsAndFilterIgnoredKeys(
+  ignoredKeys: string[],
+  _objA: Record<string, unknown>,
+  _objB: Record<string, unknown>
+): boolean {
+  // Remove ignored keys from copied objects.
+  const filteredA = deleteIgnoredKeys(ignoredKeys, _objA);
+  const filteredB = deleteIgnoredKeys(ignoredKeys, _objB);
+
+  // Compare objects without the ignored keys.
+  return lodash.isEqual(filteredA, filteredB);
+}
+
+export function compareArrayResultsWithIgnoredKeys(ignoredKeys: string[], objA: unknown[], objB: unknown[]): boolean {
+  // Remove ignored keys from each element of copied arrays.
+  const filteredA = objA?.map((obj) => deleteIgnoredKeys(ignoredKeys, obj as Record<string, unknown>));
+  const filteredB = objB?.map((obj) => deleteIgnoredKeys(ignoredKeys, obj as Record<string, unknown>));
+
+  // Compare objects without the ignored keys.
+  return isDefined(filteredA) && isDefined(filteredB) && lodash.isEqual(filteredA, filteredB);
+}
```

Comment:
Do these have general usefulness outside of the providers? If not then should we just drop them in `src/providers/utils` ?

Reply:
https://github.com/across-protocol/sdk/pull/693/commits/75528faa95972ff871ff0f36f28fafe7c814165e